internal/controller/server/client_services.go

```diff
@@ -0,0 +1,76 @@
+package server
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"octavius/internal/controller/server/execution"
+	"octavius/internal/pkg/idgen"
+	"octavius/internal/pkg/log"
+
+	octerr "octavius/internal/pkg/errors"
+	clientCPproto "octavius/internal/pkg/protofiles/client_CP"
+)
+
+type customCTXKey string
+
+var uidKey customCTXKey = "uid"
+
+type clientCPServicesServer struct {
+	procExec execution.Execution
+}
+
+// NewProcServiceServer used to create a new execution context
+func NewProcServiceServer(exec execution.Execution) clientCPproto.ClientCPServicesServer {
+	return &clientCPServicesServer{
+		procExec: exec,
+	}
+}
+
+func (s *clientCPServicesServer) PostMetadata(ctx context.Context, request *clientCPproto.RequestToPostMetadata) (*clientCPproto.MetadataName, error) {
+	uid, err := idgen.NextID()
+	if err != nil {
+		log.Error(err, "error while assigning id to the request")
+	}
+	ctx = context.WithValue(ctx, uidKey, uid)
+	log.Info(fmt.Sprintf("request ID: %v, PostMetadata request received", uid))
+	name, err := s.procExec.SaveMetadata(ctx, request.Metadata)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("request ID: %v, error in saving to etcd", uid))
+	}
+	return name, err
+}
+
+func (s *clientCPServicesServer) GetAllMetadata(ctx context.Context, request *clientCPproto.RequestToGetAllMetadata) (*clientCPproto.MetadataArray, error) {
+	uid, err := idgen.NextID()
+	if err != nil {
+		log.Error(err, "error while assigning id to the request")
+	}
+	ctx = context.WithValue(ctx, uidKey, uid)
+	log.Info(fmt.Sprintf("request ID: %v, GetAllMetadata request received", uid))
+	dataList, err := s.procExec.ReadAllMetadata(ctx)
+	return dataList, err
+}
+
+func (s *clientCPServicesServer) GetStreamLogs(request *clientCPproto.RequestForStreamLog, stream clientCPproto.ClientCPServices_GetStreamLogsServer) error {
+	uid, err := idgen.NextID()
+	if err != nil {
+		log.Error(err, "Error while assigning is to the request")
```

Comment:
this log is unnecessary. is this the top-level function that will send GRPC response to the client? I think let the client receive the error instead of logged it here.

return the error here. because this will caused the flow to still continue to below.